source/contribute.rst

```diff
@@ -41,128 +41,144 @@ contributing back to a Doctrine project.
+- If a replacement feature and/or alternative API will be provided, it
+  must be available in the minor release that adds the deprecation for
+  the old feature.
+
+With that in mind:
+
+Things that can go on the patch release branch include:
+
+- bugfixes;
+- adding tests, especially for bugs that were fixed;
+- updates, corrections or improvements to non-code assets like
+  documentation, build scripts or tooling configuration;
+- fixes to incorrect phpdoc comments (docblock type declarations, etc.);
+- updates, corrections or improvements to code comments that are not
+  phpdoc comments;
+
+.. note::
+
+   When phpdoc comments are imprecise but not wrong technically, target
+   the next minor release branch instead.
+
+The next minor version branch may include:
+
+- refactorings, unless they are necessary for a bugfix. This is to avoid
+  unnecessary risks.
+- new deprecations - remember to update the ``UPGRADE.md`` file as well!
+- adding new features and/or public APIs;
+
+On the next major version branch, usually only deprecations notices,
+deprecated features and necessary compatibility layers are being
+removed.
+
+Avoid suprising changes in public APIs on the next major version branch
+if anyhow possible. In order to provide a smooth upgrade path to users,
+the preferred solution is to add deprecations and possibly alternative
+APIs in a preceding minor version, and only remove the deprecations and
+compatibility layers in the next major release.
+
+Have you made your choice? Good. You now need to create a topic branch.
 
-.. code-block:: console
-
-    $ git clone git@github.com:username/dbal.git
-
--  Enter the dbal directory and add the **doctrine** remote
-
-.. code-block:: console
-
-    $ cd dbal
-    $ git remote add doctrine git://github.com/doctrine/dbal.git
-
-Branching from the default branch
----------------------------------
-
-New pull requests are created with the repository's default branch as base branch.
-The default branch is the branch you see when you enter the repository page on GitHub.
-
-.. image:: ../images/default-branch.png
-   :alt: The default branch
-   :style: margin-bottom: 20px
-
-In this DBAL example, it's the branch with the name **2.11.x**. The branch name reflects the
-current lowest supported version of a repository.
-
-Newly introduced changes to 2.11.x will be up-merged at a later point in time
-to newer version branches (e.g. 2.12.x, 3.0.x). This way you don't have to
-re-introduce a new fix or feature of 2.11.x with another pull request to
-the other version branches.
-
-Keeping the default branch up-to-date!
---------------------------------------
+Creating a topic branch
+-----------------------
 
-Once all this is done, you'll be able to keep your local branches up to
-date with the following command:
+Time might have elapsed since the last time you contributed or since you
+cloned the repository. You might want to fetch the latest changes from
+the ``doctrine`` remote:
```

Comment:
Maybe it’s because we removed an initial section, but nothing in the above text supports the assumption that the remote might be called like this